mantidproject/mantid
# ALC Period Info Widget

**Description of work.**
Last release a new table widget was added to the muon interfaces which displays period information from the currently loaded run. This PR implements the same table for the ALC interface too. Since ALC is written in C++ and MA/FDA in python, the widget was rewritten in C++ along with it's tests and uses sip to export the widget for use in the python GUI's

**To test:**
Open ALC interface
Load HIFI 110542
Click the periods button
There should be 8 periods with a table containing data similar to this (taken from Wimda)
![image](https://user-images.githubusercontent.com/46685490/124495371-ee2eec80-ddaf-11eb-93b5-0805502abda1.png)
The choice of columns was already decided in the previous [PR](https://github.com/mantidproject/mantid/pull/30939)
The same table should be reproducible in Muon Analysis and Frequency Domain Analysis.
Go to the grouping tab in both interface and click the periods button
Do some stress testing such as clicking the button with no data, load new data with the table open, clear the MA/FDA Guis with it open etc.

Fixes #31798 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
